ddl/ddl_db_test.go

```diff
@@ -1093,3 +1093,21 @@ out:
 	expected := fmt.Sprintf("%d %d", updateCnt, 3)
 	s.tk.MustQuery("select c2, c3 from tnn where c1 = 99").Check(testkit.Rows(expected))
 }
+
+func (s *testDBSuite) TestIssue2858And2717(c *C) {
+	defer testleak.AfterTest(c)()
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use " + s.schemaName)
+
+	s.tk.MustExec("create table t_issue_2858_bit (a bit(64) default b'0')")
+	s.tk.MustExec("insert into t_issue_2858_bit value ()")
+	s.tk.MustExec(`insert into t_issue_2858_bit values (100), ('10'), ('\0')`)
```

Comment:
add test for 
insert into t_issue_2858_bit  value('\x01')
select a + 0 from t;

In MySQL, we get: 7876657

It seems we will get 2 in TiDB with this commmit?